<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="1" passed="1" failed="0" skipped="0">
  <reporter-output>
    <line>
      <![CDATA[Hello]]>
    </line>
  </reporter-output>
  <suite started-at="2022-04-16T06:33:49 IST" name="Sample Suite" finished-at="2022-04-16T06:34:04 IST" duration-ms="14174">
    <groups>
    </groups>
    <test started-at="2022-04-16T06:33:49 IST" name="Basic Test" finished-at="2022-04-16T06:34:04 IST" duration-ms="14174">
      <class name="basePackage.BaseTest">
        <test-method is-config="true" signature="beforeTest()[pri:0, instance:basePackage.BaseTest@53ce1329]" started-at="2022-04-16T06:33:49 IST" name="beforeTest" finished-at="2022-04-16T06:34:03 IST" duration-ms="14136" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeTest -->
        <test-method is-config="true" signature="afterTest()[pri:0, instance:basePackage.BaseTest@53ce1329]" started-at="2022-04-16T06:34:04 IST" name="afterTest" finished-at="2022-04-16T06:34:04 IST" duration-ms="0" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "org.openqa.selenium.WebDriver.quit()" because "basePackage.BaseTest.driver" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "org.openqa.selenium.WebDriver.quit()" because "basePackage.BaseTest.driver" is null
at basePackage.BaseTest.afterTest(BaseTest.java:27)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:65)
at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:381)
at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:319)
at org.testng.TestRunner.invokeTestConfigurations(TestRunner.java:645)
at org.testng.TestRunner.afterRun(TestRunner.java:893)
at org.testng.TestRunner.run(TestRunner.java:606)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterTest -->
      </class> <!-- basePackage.BaseTest -->
      <class name="basePackage.TestSample">
        <test-method signature="urlValidation()[pri:0, instance:basePackage.TestSample@6283d8b8]" started-at="2022-04-16T06:34:04 IST" name="urlValidation" finished-at="2022-04-16T06:34:04 IST" duration-ms="4" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[Hello]]>
            </line>
          </reporter-output>
        </test-method> <!-- urlValidation -->
      </class> <!-- basePackage.TestSample -->
    </test> <!-- Basic Test -->
  </suite> <!-- Sample Suite -->
</testng-results>
